### DISCUSSION TITLE
[NEWBIE] - Is there a way to export a selected element into a PDF ?

### DISCUSSION THREAD

**Discussion by UnderMasterGM** [cwapi3d/cwapi3dpython#177] [Q&A]
Hello, 

I started trying to develop plugins for myself in cadwork using cwapi3d step by step. 

However I would like to know if there is a way to render an isometric .pdf view of this element ? Is this even possible ? 

I tried using FPDF and export_export_solid_with_clipboard(clipboard_number, element_id_list), but I am not sure this is the right way to go. 

Any advice ? Thanks !

**Comment by Brunner246** [comment]
Hei,
It is not possible to export to PDF.

But you can export views to images.

```python
import utility_controller as uc
import cadwork
import base64
from PIL import Image
# or use PyQt instead of PIL
# from PyQt5.QtGui import QImage 

snapshot_base64 = uc.create_snapshot("PNG")
image_bytes = base64.b64decode(snapshot_base64)
image = Image.open(io.BytesIO(image_bytes))
# PyQt
# image = QImage.fromData(image_bytes)
image.save(r"C:/Users/MichaelBrunner/Downloads/output.png")
```

**Reply by UnderMasterGM** [reply]
Thanks a lot, 

I guess i could create a new scene with this active element,
Then use vc.zoom_active_elements()
And then export the following element to an image.

The code you sent is more complex than uc.export_screen_to_image(file_path, factor), why ? 

Sorry for all the questions, I'm really a beginner and trying to get better. 

Also, I tried making a nice looking UI, using Tkinter, but I cannot use a background image ? Is there a way around ?

**Comment by Brunner246** [comment]
`export_screen_to_image` exports the data in `.TIF` format. 
With `create_snapshot` you can export to `.JPG` , `.PNG` , ...

https://www.geeksforgeeks.org/how-to-use-images-as-backgrounds-in-tkinter/

**Reply by UnderMasterGM** [reply]
Thanks. 

Tell me if I am wrong, but I can't get Tkinter to work with a background when I'm launching it from cadwork until I use the IDLE Shell. 
Really strange.

**Reply by Brunner246** [reply]
Please post a code snippet so that i can reproduce that.

**Reply by ancelin-charpentier** [reply]
Hey @UnderMasterGM,
I am a newbie too but I had to fix this one as well.
I got it fixed by relocating the "lib" folder and all its content to this location:

C:\Program Files\cadwork.dir\EXE_30\lib

And then it worked.

The "lib" was not saved in the right location, you can find it by searching for lib in your cadwork.dir 

I hope that will help you.
